enhancements/installer/azure-support-known-cloud-environments.md

```diff
@@ -0,0 +1,231 @@
+---
+title: azure-support-known-cloud-environments
+authors:
+  - "@abhinavdahiya"
+reviewers:
+  - TBD
+creation-date: 2020-05-11
+last-updated: 2020-05-11
+status: implementable
+see-also:
+  - "/enhancements/installer/aws-custom-region-and-endpoints.md"  
+replaces:
+superseded-by:
+---
+
+# Azure Support Known Cloud Environments
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As an administrator, I would like to deploy OpenShift 4 clusters to non-public Azure clouds. There are various known Azure clouds in addition to the `AzurePublicCloud` available to users, including but not limited to `AzureUSGovernmentCloud` etc. and requires the cluster to use different Azure API endpoints. The user should be allowed to provide the name of the Azure cloud and the installer and the cluster operators should use the correct API endpoints for that provided cloud.
```

Comment:
can we include a link to azure docs for those "various known Azure clouds"?

Reply:
added to the goals.